docker-compose.services.yml

```diff
@@ -88,9 +88,58 @@ services:
     profiles:
       - load-testing
 
+  keycloak:
+    profiles:
+      - keycloak
+    image: quay.io/keycloak/keycloak:latest
+    depends_on:
+      db:
+        condition: service_healthy
+    ports:
+      - ${KEYCLOAK_PORT}:${KEYCLOAK_PORT}
+      - ${KEYCLOAK_SSL_PORT}:${KEYCLOAK_SSL_PORT}
+    environment:
+      - KEYCLOAK_ADMIN=${KEYCLOAK_SVC_ADMIN:-admin}
+      - KEYCLOAK_ADMIN_PASSWORD=${KEYCLOAK_SVC_ADMIN_PASSWORD:-admin}
+    networks:
+      default:
+        aliases:
+          - ${KEYCLOAK_SVC_HOSTNAME:-kc.ol.local}
+    command: start --verbose --features scripts --import-realm --hostname=${KEYCLOAK_SVC_HOSTNAME:-kc.ol.local} --hostname-strict=false --hostname-debug=true --https-port=${KEYCLOAK_SSL_PORT} --https-certificate-file=/etc/x509/https/tls.crt --https-certificate-key-file=/etc/x509/https/tls.key --http-enabled=true --http-port=${KEYCLOAK_PORT} --config-keystore=/etc/keycloak-store --config-keystore-password=${KEYCLOAK_SVC_KEYSTORE_PASSWORD} --db=postgres --db-url-database=keycloak --db-url-host=db --db-schema=public --db-password=${POSTGRES_PASSWORD:-postgres} --db-username=postgres --db-url-port=${PGPORT:-5432}
+    volumes:
+      - keycloak-store:/etc/keycloak-store
+      - ./config/keycloak/tls:/etc/x509/https
+      - ./config/keycloak/realms:/opt/keycloak/data/import
+      - ./config/keycloak/providers:/opt/keycloak/providers
+      - ./config/keycloak/themes:/opt/jboss/keycloak/themes
+
+  apigateway:
+    profiles:
+      - keycloak
+    image: apache/apisix:latest
+    depends_on:
+      - keycloak
```

Comment:
I would remove `profiles` and `depends_on` here, because otherwise I'm forced to spend resources running a duplicate keycloak instance I won't use if I'm not running the keycloak container defined above.

Reply:
Might it be useful to still keep the `profiles` part, but change it to its own value so that apisix (and keycloak) don't start automatically, only when explicitly requested? So:

`COMPOSE_PROFILES=backend,frontend,keycloak,apisix` starts everything
`COMPOSE_PROFILES=backend,frontend,apisix` starts apisix but not keycloak
etc

Reply:
I just made that change (removed `depends_on`,  gave keycloak and apisix containers their own separate profiles)

Reply:
Forgot you can just call them by name in this case, so updated again